File: node/routers/SearchRouter.js
const express = require('express');
const router = express.Router();

const SearchSongs = require('../controllers/searchSongs');
const KeywordProvider = require('../utils/KeywordProvider');

router.get('/', async (req, res) => {
  try {
    const artist = req.query.artist || '';
    const title = req.query.title || '';
    const keyword=KeywordProvider.getKeyword(artist,title);
    const connection = req.app.locals.connection;

    //Genius API
    const searchSongs=new SearchSongs();
    songInfoList= await searchSongs.getSongListFromApi(keyword);

    if (songInfoList.length === 0) {
        return res.status(404).json({ error: 'No songs found.' });
      }
      
    return res.json(songInfoList);
  } catch (error) {
    console.error(error);
    return res.status(500).json({ error: 'Internal server error.' });
  }
});

module.exports = router;

File: app/routers/SearchRouter.py
from fastapi import APIRouter, Query, HTTPException, Depends
from typing import List
from pydantic import BaseModel

# 必要なインポート
from controllers.genius_api import get_songs_from_genius_api
from utils.KeywordProvider import get_keyword

router = APIRouter()


# Pydanticモデル
class SongInfo(BaseModel):
    artist: str
    title: str
    songUrl: str
    imageUrl: str
    keyword: str


@router.get("")
def get_song_list(
    artist: str = Query("", alias="artist"), title: str = Query("", alias="title")
):
    # キーワードを取得
    keyword = get_keyword(artist, title)

    # Genius APIから曲リストを取得
    song_info_list = get_songs_from_genius_api(keyword)

    if not song_info_list:
        raise HTTPException(status_code=404, detail="No songs found.")

    return song_info_list
